sql/analyzer/validation_rules.go

```diff
@@ -270,6 +309,25 @@ func expressionReferencesOnlyGroupBys(groupBys []string, expr sql.Expression) bo
 			if stringContains(groupBys, expr.String()) {
 				return false
 			}
+			return true
+		case *expression.GetField:
+			for _, groupBy := range groupBys {
```

Comment:
You can call stringContains for both these checks, rather than looping over groupBys